torch/csrc/autograd/python_torch_functions_manual.cpp

```diff
@@ -393,13 +393,12 @@ static PyObject* THPVariable__mirror_autograd_meta_to(
   auto r = parser.parse(args, kwargs, parsed_args);
   auto src_ = r.tensor(0);
   auto dst_ = r.tensor(1);
-  // Here, we unsafely set the grad function on the wrapper to be the same as
-  // the inner. We expect this grad_fn to NEVER be used. It's needed so that
-  // .is_leaf metadata is accurate on the wrapper
   auto inner_autograd_meta = impl::get_autograd_meta(src_);
   if (inner_autograd_meta) {
     dst_.set_requires_grad(src_.requires_grad());
-    if (dst_.requires_grad()) {
+    if (inner_autograd_meta->grad_fn_) {
```

Comment:
This feels reasonable but wondering if Alban/Jeffrey have any thoughts

@yf225 I'm also curious how you hit this situation - is there some repro code using torch.compile that needs it?

Reply:
I hit this when doing FSDP tracing - I tried to call register_post_accumulate_grad_hook on a requires_grad=True grad_fn=None leaf tensor, and it complains that I can't register post-acc-grad hook on a "non-leaf tensor" even though this tensor is actually leaf.

Let me add a test case for this.

Reply:
I feel like you have to call src_->grad_fn() here.
The code there is written such that a grad_fn_ might appear during that call. I'm not sure off the top of my head that is the exact set of operations you need to do to get there but we shouldn't assume the grad_fn_ field is up to date.